langchain/src/agents/agent.ts

```diff
@@ -219,12 +220,15 @@ export abstract class Agent {
   /**
    * Load an agent from a json-like object describing it.
    */
-  static async deserialize(
-    data: SerializedAgent & { llm?: BaseLLM; tools?: Tool[] }
+  static async deserialize<T extends string, U, V extends AgentInput>(
```

Comment:
Should an agent know anything about its implementations? could/should this just be an abstract method?